tracer/build/_build/Build.GitHub.cs

```diff
@@ -346,60 +394,89 @@ partial class Build
             };
         });
 
-    Target ExtractReleaseNotes => _ => _
-      .Unlisted()
-      .Executes(() =>
-       {
-           Console.WriteLine("Reading changelog...");
-
-           var changelogPath = RootDirectory / "docs" / "CHANGELOG.md";
-           var changelog = File.ReadAllText(changelogPath);
-
-           // find first header
-           var releaseNotesStart = changelog.IndexOf($"## [Release {FullVersion}]");
-           var firstContent = changelog.IndexOf("##", startIndex: releaseNotesStart + 3);
-           var releaseNotesEnd = changelog.IndexOf($"## [Release", firstContent);
-
-           var artifactsLink = Environment.GetEnvironmentVariable("PIPELINE_ARTIFACTS_LINK");
-
-           var sb = new StringBuilder();
-           sb.AppendLine($"⚠ 1. Download the NuGet packages for the release from [this link]({artifactsLink}) and upload to nuget.org");
-           sb.AppendLine("⚠ 2. Download the signed MSI assets and native symbols from GitLab and attach to this release before publishing");
-           sb.AppendLine();
-           sb.Append(changelog, firstContent, releaseNotesEnd - firstContent);
-
-           Console.WriteLine(sb.ToString());
-
-           // need to encode the release notes for use by github actions
-           // see https://trstringer.com/github-actions-multiline-strings/
-           sb.Replace("%","%25");
-           sb.Replace("\n","%0A");
-           sb.Replace("\r","%0D");
-
-           Console.WriteLine("::set-output name=release_notes::" + sb.ToString());
-
-           Console.WriteLine("Release notes generated");
-       });
-
-
     Target DownloadAzurePipelineArtifacts => _ => _
        .Unlisted()
+       .Description("Downloads the latest artifacts from Azure Devops that has the provided version")
        .DependsOn(CreateRequiredDirectories)
        .Requires(() => AzureDevopsToken)
        .Requires(() => Version)
+       .Requires(() => TargetBranch)
        .Executes(async () =>
        {
             // Connect to Azure DevOps Services
             var connection = new VssConnection(
                 new Uri(AzureDevopsOrganisation),
                 new VssBasicCredential(string.Empty, AzureDevopsToken));
 
-            // Get a GitHttpClient to talk to the Git endpoints
+            // Get an Azure devops client
             using var buildHttpClient = connection.GetClient<BuildHttpClient>();
 
-            var branch = $"refs/tags/v{FullVersion}";
+            // Get all the builds to TargetBranch that were triggered by a CI push
+
+            var builds = await buildHttpClient.GetBuildsAsync(
```

Comment:
With the issue that we had on Friday (ie no builds running against master), there's a non zero risk that we actually fetch the wrong build. I don't think it will happen but still it can happen. 
I still like this approach but I think we should find ways to be alerted that CI is not running on master.

Reply:
> I still like this approach but I think we should find ways to be alerted when CI is not running on master.

I agree. It would be good to be able to easily detect if a commit _should_ have a build. Basically means creating our own yaml pipeline trigger parsing action though, unless you have a better idea?

Reply:
I have no idea yet, if I had I would have suggested it here :) I'll think about it. 
If the source of truth is Github, we could have actions on merge on Github that checks that the no-op or consolidated pipeline is running against master on this commit. (but I still need to tjink about it :p)